### PR TITLE
Allow to set per cf ttl

### PR DESCRIPTION
**Pull request by 0xdeafbeef**

### CODE REVIEW

File: src/column_family.rs [hunk 1]
```diff
@@ -28,24 +29,64 @@ pub const DEFAULT_COLUMN_FAMILY_NAME: &str = "default";
...
+
+    pub fn ttl(&self) -> CfTtl {
+        self.ttl
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+/// Specifies the TTL behavior for a column family.
+/// https://github.com/facebook/rocksdb/blob/18cecb9c46b4c2a8b148659dac2fcab5a843d32b/include/rocksdb/utilities/db_ttl.h#L16-L46
+pub enum CfTtl {
```
Comment (aleksuss) [inline src/column_family.rs:83]: I would suggest to rename to the `ColumnFamilyTtl` to be consistent. E.g. `ColumnFamilyDescriptor`

Comment (0xdeafbeef) [reply src/column_family.rs]: Done

File: src/column_family.rs [hunk 2]
```diff
@@ -28,24 +29,64 @@ pub const DEFAULT_COLUMN_FAMILY_NAME: &str = "default";
...
+    {
+        Self {
+            name: name.into(),
+            options,
+            ttl,
+        }
+    }
+
+    /// Sets ttl for the column family. It's applied only when DB is opened with
+    /// [`DB::open_with_ttl()`]. Changing ttl after DB is opened has no effect.
```
Comment (aleksuss) [inline src/column_family.rs:65]:
```suggestion
    /// [`crate::db::DB::open_with_ttl()`]. Changing ttl after DB is opened has no effect.
```

**Comment by 0xdeafbeef** [comment]
fixes #958

File: src/column_family.rs [hunk 3]
```diff
@@ -28,24 +29,64 @@ pub const DEFAULT_COLUMN_FAMILY_NAME: &str = "default";
...
+/// Specifies the TTL behavior for a column family.
+/// <https://github.com/facebook/rocksdb/blob/18cecb9c46b4c2a8b148659dac2fcab5a843d32b/include/rocksdb/utilities/db_ttl.h#L16-L46>
+pub enum ColumnFamilyTtl {
+    /// Will internally set TTL to -1 (disabled)
+    Disabled,
+    /// Will set ttl to the specified duration
+    Set(Duration),
+    /// Will use ttl specified at db open time
+    SameAsWholeDb,
 }
```
Comment (aleksuss) [inline src/column_family.rs:90]: A couple of suggestions here. Looks like it makes sense to add the `Default` trait to the enum.  And I would rename the `Set` variant to `Duration`. It's a rust idiomatic, I guess. 
```rust
#[derive(Debug, Default, Clone, Copy)]
 /// Specifies the TTL behavior for a column family.
 /// <https://github.com/facebook/rocksdb/blob/18cecb9c46b4c2a8b148659dac2fcab5a843d32b/include/rocksdb/utilities/db_ttl.h#L16-L46>
 pub enum ColumnFamilyTtl {
     /// Will internally set TTL to -1 (disabled)
     #[default]
     Disabled,
     /// Will set ttl to the specified duration
     Duration(Duration),
     /// Will use ttl specified at db open time
     SameAsDb,
 }
```

Comment (aleksuss) [reply src/column_family.rs:90]: By the way, I'm going to include this PR to #964 and I can do it by myself if you don't mind.

Comment (0xdeafbeef) [reply src/column_family.rs:91]: Maybe InheritDbTtl?

**Review by aleksuss — APPROVED** [review]
Good work 👍🏻

**Review by stanislav-tkach — APPROVED** [review]

File: src/column_family.rs (hunk 3 above)
Comment (aleksuss) [reply src/column_family.rs:91]: > Naming is hard :)
Sometimes it is 😄

But as for me, whether `SameAsWholeDb` or  `SameAsDb` is fine.

**Pull request merged**